adapters/faraday/example/faraday_no_op.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+# Copyright 2020 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'rubygems'
+require 'bundler/setup'
+
+Bundler.require
+
+require_relative '../lib/opentelemetry/adapters/faraday/middlewares/tracer_middleware'
+# Demonstrate disabling span reporting
+class NoOp < OpenTelemetry::Adapters::Faraday::Middlewares::TracerMiddleware
+  def disable_span_reporting?(env)
+    env.url.to_s =~ /example.com/
+  end
+end
+
+OpenTelemetry::SDK.configure do |c|
+  c.use 'OpenTelemetry::Adapters::Faraday', tracer_middleware: NoOp
```

Comment:
@mwear @elskwid Based on our discussion, I've updated the example files to use the proper installation and to have the no-op examples in separate files.